lib-rs/src/lib.rs

```diff
@@ -27,7 +109,34 @@ pub unsafe fn slice_from_raw_parts<'a, T>(ptr: *const T, len: usize) -> &'a [T]
     }
 }
 
-/// Wraps a non-owning FFI string
+/// Wraps a non-owning FFI string without the overhead of CString/String
+///
+/// This allows the user of the API to choose if they want to treat the string
+/// as a bunch of bytes (`bytes`), verify it as a utf8 `str` (`as_str`), or
+/// convert it lossily into a `Cow<str>` (`to_str_lossy`).
```

Comment:
I would suggest creating links for the method names.

```suggestion
/// as a bunch of bytes ([`bytes`]), verify it as a utf8 `str` ([`as_str`]), or
/// convert it lossily into a `Cow<str>` ([`to_str_lossy`]).
```